File: app/src/main/java/br/com/alura/ceep/ui/coffemachine/presentation/view/CoffesRepository.java
package br.com.alura.ceep.ui.coffemachine.presentation.view;

import android.app.Application;
import android.os.AsyncTask;

import androidx.lifecycle.LiveData;

import java.util.List;

public class CoffesRepository {
    private CoffesDao mnamesDao;
    private CoffesDao mquantityDao;
    private LiveData<List<Coffes>> mAllNames;
    private LiveData<List<Coffes>> mAllCapsules;

    CoffesRepository (Application application) {
        CoffeRoomDataBase db = CoffeRoomDataBase.getDatabase (application);
        mnamesDao = db.coffesDao ();
        mquantityDao = db.coffesDao ();
        mAllNames = mnamesDao.getAllNames ();
        mAllCapsules = mquantityDao.getAllCapsules ();
    }

    LiveData<List<Coffes>> getAllNames () {
        return mAllNames;
    }

    LiveData<List<Coffes>> getAllCapsules () {
        return mAllCapsules;
    }

    public void insert (Coffes coffes) {
        new insertAsyncTask (mnamesDao).execute (coffes);
    }

    public void insert (Coffes coffes) {
        new insertAsyncTask (mquantityDao).execute (coffes);
    }


    private static class insertAsyncTask extends AsyncTask <Coffes, Void, Void>{
        private CoffesDao mAsyncTaskDao;

        insertAsyncTask(CoffesDao dao) {
            mAsyncTaskDao = dao;
        }
        public void execute (Coffes coffes) {
        }
        @Override
        protected Void doInBackground(final Coffes... params) {
            mAsyncTaskDao.insert(params[0]);
            return null;
        }
    }
}

File: app/src/main/java/br/com/alura/ceep/ui/coffemachine/presentation/view/CoffesRepository.kt
package br.com.alura.ceep.ui.coffemachine.presentation.view

class CoffesRepository(private val coffesDao: CoffesDao) {

    fun get() = coffesDao.getAllCoffes()

    fun getSpecicCoffe(id: Long) =
        coffesDao.getById(id)


    fun delete(coffes: Coffes): Boolean {
        val id = coffesDao.deleteCoffe(coffes)
        return id > 0
    }

    fun save(coffes: Coffes): Boolean {
        val id = coffesDao.save(coffes)
        return id > 0
    }


//    private val mnamesDao: CoffesDao
//    private val mquantityDao: CoffesDao
//    val allNames: LiveData<List<Coffes>>
//    val allCapsules: LiveData<List<Coffes>>
//
//    fun insert(coffes: Coffes?) {
//        insertAsyncTask(mnamesDao).execute(coffes)
//    }
//
//    fun insert(coffes: Coffes?) {
//        insertAsyncTask(mquantityDao).execute(coffes)
//    }
//
//    private class insertAsyncTask internal constructor(private val mAsyncTaskDao: CoffesDao) :
//        AsyncTask<Coffes?, Void?, Void?>() {
//        fun execute(coffes: Coffes?) {}
//        protected override fun doInBackground(vararg params: Coffes): Void? {
//            mAsyncTaskDao.insert(params[0])
//            return null
//        }
//    }

//    init {
//        val db = CoffeRoomDataBase.getDatabase(application)
//        mnamesDao = db.coffesDao()
//        mquantityDao = db.coffesDao()
//        allNames = mnamesDao.allNames
//        allCapsules = mquantityDao.allCapsules
//    }
}